src/components/Grid/AgGrid/AgGrid.tsx

```diff
@@ -0,0 +1,276 @@
+            // destruct doesn't include functions, so we need to redeclare them
+            //apiListeners: agGridRef.current?.apiListeners,
+            registerApiListener: agGridRef.current?.registerApiListener,
+            //setGridApi: agGridRef.current?.setGridApi,
+            componentWillUnmount: agGridRef.current?.componentWillUnmount,
+            render: agGridRef.current?.render,
+            setState: agGridRef.current?.setState,
+            forceUpdate: agGridRef.current?.forceUpdate,
+            //...
+            context: agGridRef.current?.context as AgGridRef<
+                TData,
+                TContext
+            >['context'],
+        }),
+        []
+    );
+
+    const translations = useMemo(
+        () => messages[intl.locale] ?? messages[intl.defaultLocale],
+        [intl.locale, intl.defaultLocale]
+    );
+
+    const customTheme = useMemo(
+        () =>
+            deepmerge(
+                {
+                    width: '100%',
+                    height: '100%',
+                    //@media print -> page-break-inside: avoid
+                },
+                deepmerge(theme.agGridThemeOverride ?? {}, {
+                    '--ag-icon-font-family': 'agGridMaterial !important',
+                    '& *': {
+                        '--ag-icon-font-family': 'agGridMaterial !important',
+                    },
+                })
+            ),
+        [theme.agGridThemeOverride]
+    );
+    /*const customContext = useMemo(
+        () =>
+            ({
+                ...(props.context ?? ({} as TContext)),
+                //TODO
+            } as TContext),
+        [props.context]
+    );*/
+
+    return (
+        // wrapping container with theme & size
+        <Box component="div" className={theme.agGridTheme} sx={customTheme}>
+            <AgGridReact<TData>
+                ref={agGridRef}
+                modules={[
+                    //ClientSideRowModelModule,
+                    CsvExportModule,
+                ]}
+                localeText={translations}
+                /*getLocaleText={(params) =>
+                    intl.formatMessage(
+                        { id: params.key, defaultMessage: params.defaultValue },
+                        Array.reduce(
+                            params.variableValues ?? [],
+                            (acc, value, idx, array) => ({
+                                ...acc,
+                                [`var${idx}`]: value,
+                            }),
+                            {}
+                        )
+                    )
+                }*/
+                {...props} //destruct props to optimize react props change detection
+                debug={
+                    process.env.REACT_APP_DEBUG_AGGRID === 'true' || props.debug
+                }
+                //context={customContext}
+                defaultCsvExportParams={useSecuredCsvExportParams(
```

Comment:
In my opinion we should remove this code as CSV export is not enabled and it's not consistent with our other Aggrid tables